cartographer/mapping/sparse_pose_graph.cc

```diff
@@ -84,22 +95,39 @@ proto::SparsePoseGraph SparsePoseGraph::ToProto() {
     Eigen::Map<Eigen::Matrix<double, 6, 6>>(
         constraint_proto->mutable_sqrt_lambda()->mutable_data()) =
         constraint.pose.sqrt_Lambda_ij;
-    // TODO(whess): Support multi-trajectory.
+
     constraint_proto->mutable_submap_id()->set_submap_index(constraint.i);
```

Comment:
These indexes are still into the global list of nodes/submaps. I think you need two counters for each trajectory to remap the indexes to 0, 1, ... for each trajectory.